src/lib/efhw/efct.c

```diff
@@ -107,6 +122,39 @@ efct_get_hugepages(struct efhw_nic *nic, int hwqid,
   return rc;
 }
 
+static ssize_t
+efct_get_used_hugepages(struct efhw_nic *nic, int qid)
+{
+  struct xlnx_efct_hugepage *pages;
+  ssize_t used;
+  int i, rc;
+
+  pages = kvzalloc(sizeof(pages[0]) * CI_EFCT_MAX_HUGEPAGES, GFP_KERNEL);
+  if( ! pages )
+    return -ENOMEM;
+
+  /* This call can return `EINVAL` for a few reasons, including the case where
+   * the provided `qid` is not bound to by `nic`. Instead of returning an error
+   * here, we should instead validly claim that no hugepages are being used. */
```

Comment:
Did I complain about this before? Can you remember what I said?

Reply:
I think you might have previously, but I believe I addressed the issues you raised. I have a feeling it was because I made certain assumptions about what the X3 driver does if `rc == -EINVAL` and just fell through to the for loop because the allocation would still be valid (and zeroed) resulting in `used = 0`. Instead, we just return 0 early instead.